src/Symfony/Component/Form/Extension/Core/EventListener/ResizeFormListener.php

```diff
@@ -110,6 +116,29 @@ public function preSubmit(FormEvent $event)
             throw new UnexpectedTypeException($data, 'array or (\Traversable and \ArrayAccess)');
         }
 
+        // Fix array indices
+        if ($this->reindex) {
+            $originalData = $data;
```

Comment:
What if $originalData is not an array? At the moment, this method also supports array-like objects, although I don't understand why. The check was introduced in 77bea81ed5d9513cd32e5ad2a530e945832ae3b9, but I don't know why anymore why. Any ideas @stof @Tobion?